test/unittests/execution_state_test.cpp

```diff
@@ -138,30 +138,29 @@ TEST(execution_state, reset_advanced)
     }
 }
 
-TEST(execution_state, stack_clear)
+TEST(execution_state, stack_reset)
 {
     evmone::Stack stack;
+    intx::uint256 stack_space[2]{};
+    stack.reset(stack_space);
 
-    stack.clear();
     EXPECT_EQ(stack.size(), 0);
-    EXPECT_EQ(stack.top_item + 1, stack.storage);
```

Comment:
Why not use `StackSpace` in these teste? Then you could keep checking `stack.top_item` against `stack_space.bottom()`

Reply:
I mean can you keep here
```
EXPECT_EQ(stack.top_item + 1, stack_space.bottom());
```

Reply:
GCC complains that "array subscript -1 is outside array bounds". I gave up.